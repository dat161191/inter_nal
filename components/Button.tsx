import React from 'react';
import { Spinner } from 'reactstrap';

interface IButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  isLoading?: boolean;
  colorSpinner?: string;
}

const Button: React.FC<IButtonProps> = ({ isLoading, children, colorSpinner = 'light', ...props }: IButtonProps) => (
  <button {...props} disabled={isLoading || props.disabled} className={props.className}>
    {isLoading && <Spinner color={colorSpinner} size="sm" className="mr-2" />}
    {children}
  </button>
);

export default Button;
