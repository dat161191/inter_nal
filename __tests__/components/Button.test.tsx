import React from 'react';
import { render } from '@testing-library/react';
import Button from '../../components/Button';

const exampleInfo = {
  isLoading: true,
  colorSpinner: '#f4f4f4',
  children: 'button content',
};

describe('Button', () => {
  test('should render correctly loading', () => {
    const { container } = render(<Button {...exampleInfo} />);
    expect(container.querySelector('.sr-only')).toBeTruthy();
  });
  test('should render correctly content', () => {
    const { container, getByText } = render(<Button {...exampleInfo} isLoading={false} />);
    expect(container.querySelector('.sr-only')).toBeFalsy();
    expect(getByText(exampleInfo.children)).toBeTruthy();
  });
});
