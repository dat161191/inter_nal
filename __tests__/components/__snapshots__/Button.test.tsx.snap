// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div>
  <button
    disabled=""
  >
    <div
      class="mr-2 spinner-border-sm spinner-border text-#f4f4f4"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
    button title
  </button>
</div>
`;
